x/interchainstaking/ibc_module.go

```diff
@@ -217,6 +218,14 @@ func (im IBCModule) OnAcknowledgementPacket(
 	acknowledgement []byte,
 	relayer sdk.AccAddress,
 ) error {
+	connectionId, _, err := im.keeper.IBCKeeper.ChannelKeeper.GetChannelConnection(ctx, packet.SourcePort, packet.SourceChannel)
+	if err != nil {
+		err = fmt.Errorf("packet connection not found: %w", err)
+		ctx.Logger().Error(err.Error())
+		return err
+	}
+	ctx = ctx.WithContext(context.WithValue(ctx.Context(), "connectionId", connectionId))
```

Comment:
Is it worth doing the same thing for `chain-id`?

Reply:
I was wondering the same thing...
but we are always able to get the chainId from the connectionId (convenience methods implemented in this PR),
and seeing as it is recommended to not add unnecessary values to context it seemed best to only add the conncetionId that would otherwise be unobtainable further down the execution stack;